nx/lib/nx.ex

```diff
@@ -323,6 +323,79 @@ defmodule Nx do
   For a more complex slicing rules, including strides, you
   can always fallback to `Nx.slice/4`.
 
+  Access also supports updating values with put_in/3 and update_in/3,
+  where the path to the value is a list of integers indicating the sub-dimensions.
+
+  The value that will be used to substitute should be a tensor of the same
+  shape as the one being updated.
+
+      iex> t = Nx.tensor([1, 2, 3])
+      iex> put_in(t, [2], Nx.tensor(2))
+      #Nx.Tensor<
+        s64[3]
+        [1, 2, 2]
+      >
+
+  In a multi-dimensional array, the path will determine at which level
+  the update will be made.
+
+  A whole tensor can be updated:
+
+      iex> t = Nx.tensor([[1, 2], [3, 4]])
+      iex> put_in(t, [1], Nx.tensor([4, 5]))
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [1, 2],
+          [4, 5]
+        ]
+      >
```

Comment:
"A whole tensor can be updated" doesn't seem to be compatible with the example (in which only part of the tensor is updated)

Reply:
```suggestion
  One can use a tensor with various entries as an update value:

      iex> t = Nx.tensor([[1, 2], [3, 4]])
      iex> put_in(t, [1], Nx.tensor([4, 5]))
      #Nx.Tensor<
        s64[2][2]
        [
          [1, 2],
          [4, 5]
        ]
      >
```

Reply:
Hmm, what I'm trying to say is that the inner element which is also a tensor can be updated as a whole. I agree that what I wrote is confusing but I'm thinking if there could be another way to state it in the sense that an N-dimensional tensor can be used as the update value as long as the shape matches with the element at the level it tries to replace it.

Reply:
WDYT about how it is written now, @polvalente ? https://github.com/elixir-nx/nx/pull/864/commits/f279b5490efec83586be4876655f87cafb8c6776